gtk4/src/subclass/widget.rs

```diff
@@ -550,8 +557,15 @@ unsafe extern "C" fn widget_compute_expand<T: WidgetImpl>(
     let mut vexpand: bool = from_glib(*vexpand_ptr);
 
     imp.compute_expand(wrap.unsafe_cast_ref(), &mut hexpand, &mut vexpand);
-    *hexpand_ptr = hexpand.to_glib();
-    *vexpand_ptr = vexpand.to_glib();
+
+    let widget = wrap.unsafe_cast_ref::<Widget>();
+
+    if !widget.get_hexpand_set() {
+        *hexpand_ptr = hexpand.to_glib();
```

Comment:
This is not a problem, the C code just ignores it. You can remove this part, but ...